src/little_dict.jl

```diff
@@ -0,0 +1,89 @@
+"""
+    LittleDict(keys, vals)<:AbstractDict
+
+A ordered dictionary type for small numbers of keys.
+Rather than using `hash` or some other sophisicated measure
+to store the vals in a clever arrangement,
+it just keeps everything in a pair of lists.
+
+While theoretically this has expected time complexity _O(n)_,
+vs the hash-based `OrderDict`/`Dict`'s expected time complexity _O(1)_,
+and the search-tree-based `SortedDict`'s expected time complcity _O(log(n))_.
+In practice it is really fast, because it is cache & SIMD friendly.
+
+It is reasonable to expect it to outperform an `OrderedDict`,
+with up to around 30 elements in general;
+or with up to around 50 elements if using a `LittleDict` backed by `Tuples`
+(see [`freeze`](@ref))
+However, this depends on exactly how long `isequal` and `hash` take,
+as well as on how many hash collisions occur etc.
+"""
+struct LittleDict{K,V,KS,VS} <: AbstractDict{K, V}
+    keys::KS
+    vals::VS
```

Comment:
Do you need an inner constructor that checks that `keys` and `vals` have the same axes?

Reply:
Yes.
Though,
I am tempted to just restrict it to `Tuple` and `Vector`.
I don't see advantage of any other types,
except perhaps to avoid effectively copying the keys/values.
And that itself seems like a dangerous thing to do anyway.

Reply:
+1 to have one mutable and one "frozen" type and do away with the `KS,VS` type variables.  The other uses seem to niche to have to carry around those `KS,VS` everywhere.

Reply:
> The other uses seem to niche to have to carry around those KS,VS everywhere.

They are not heavy, and you can just ignore them.

They will still remain anyway as I currently envision it,
because it will save code. and tuples are akward to deal with as types.
In particular
would still allow halffrozen options, with `Tuple` keys, and `Vector` values.
So you can't add or remove values but you can mutated them.

Reply:
I think you still need to check that they have the same length?